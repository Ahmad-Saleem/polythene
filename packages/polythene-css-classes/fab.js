
export default {
  component: "pe-button pe-fab",

  // elements
  content:   "pe-fab__content",

  // states
  mini:      "pe-fab--mini",
};
