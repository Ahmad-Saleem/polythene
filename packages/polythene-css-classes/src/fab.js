
export const fabClasses = {
  component: "pe-fab",

  // elements
  content:   "pe-fab__content",

  // states
  mini:      "pe-fab--mini",
};
