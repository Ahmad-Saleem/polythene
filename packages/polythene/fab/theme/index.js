import styler from '../../common/styler';
import { componentConfig } from 'polythene-theme';
import vars from './config';
import layout from './layout';
import color from './color';

var configFn = componentConfig && componentConfig.fab;
var config = configFn ? configFn(vars) : vars;
var id = 'pe-fab';

styler.add(id, layout(config), color(config));