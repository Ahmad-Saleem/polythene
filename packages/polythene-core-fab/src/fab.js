import { customTheme } from "./theme";
import classes from "./classes";

export const theme = customTheme;

// Don't export 'element': that will be the wrapped raised button component (set in polythene-xxx-fab)

// Props to be passed to a raised button, including 'content'

export const createProps = (vnode, { keys: k, renderer: h, icon }) => {
  const attrs = vnode.attrs;
  const content = attrs.content
    ? attrs.content
    : attrs.icon
      ? h(icon, attrs.icon)
      : attrs.children || vnode.children;
  return Object.assign(
    {},
    {
      content: h("div", {
        className: classes.content
      }, content),
      parentClassName: [
        classes.component,
        attrs.mini ? classes.mini : null,
        attrs.className || attrs[k.class],
      ].join(" "),
      // defaults
      ripple: {
        center: true,
        opacityDecayVelocity: 0.24
      },
      shadow: { increase: 5 },
      ink: true,
      wash: true,
      animateOnTap: attrs.animateOnTap !== undefined
        ? attrs.animateOnTap
        : true
    },
    attrs
  );
};

export const createContent = () => null;
