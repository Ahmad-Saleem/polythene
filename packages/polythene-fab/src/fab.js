import m from "mithril";
import raisedButton from "polythene-raised-button";
import icon from "polythene-icon";
import { customTheme } from "./theme";

export const classes = {
  component: "pe-fab",
  content:   "pe-fab__content",
  mini:      "pe-fab--mini"
};

const view = vnode => {
  const attrs = vnode.attrs;
  const content = attrs.content
    ? attrs.content
    : attrs.icon
      ? m(icon, attrs.icon)
      : attrs.children || vnode.children;
  return m(raisedButton, Object.assign(
    {},
    {
      content: m("div", {
        class: classes.content
      }, content),
      parentClass: [
        classes.component,
        attrs.mini ? classes.mini : null
      ].join(" "),
      // defaults
      ripple: {
        center: true,
        opacityDecayVelocity: 0.24
      },
      shadow: { increase: 5 },
      ink: true,
      wash: true,
      animateOnTap: attrs.animateOnTap !== undefined
        ? attrs.animateOnTap
        : true
    },
    attrs
  ));
};

export default {
  theme: customTheme, // accepts (selector, vars)
  view
};
