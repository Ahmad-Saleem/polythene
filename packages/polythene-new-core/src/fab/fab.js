import m from "mithril";
import raisedButton from "polythene-raised-button";
import icon from "polythene-icon";
import { customTheme } from "./theme";
import classes from "./classes";

export const element = "div";

export const theme = customTheme;

export const createProps = (vnode, { keys: k }) => {
  // const attrs = vnode.attrs;
};

export const createContent = (vnode, { renderer: h }) => {
  // const attrs = vnode.attrs;
};

const view = vnode => {
  const attrs = vnode.attrs;
  const content = attrs.content
    ? attrs.content
    : attrs.icon
      ? m(icon, attrs.icon)
      : attrs.children || vnode.children;
  return m(raisedButton, Object.assign(
    {},
    {
      content: m("div", {
        class: classes.content
      }, content),
      parentClass: [
        classes.component,
        attrs.mini ? classes.mini : null
      ].join(" "),
      // defaults
      ripple: {
        center: true,
        opacityDecayVelocity: 0.24
      },
      shadow: { increase: 5 },
      ink: true,
      wash: true,
      animateOnTap: attrs.animateOnTap !== undefined
        ? attrs.animateOnTap
        : true
    },
    attrs
  ));
};

export default {
  theme: customTheme, // accepts (selector, vars)
  view
};
