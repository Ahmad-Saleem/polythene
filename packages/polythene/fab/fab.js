var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

import '../common/object.assign';
import m from 'mithril';
import iconBtn from '../icon-button/icon-button';
import './theme';

var CSS_CLASSES = {
    block: 'pe-button--fab',
    mini: 'pe-button--fab-mini'
};

var createView = function createView(ctrl) {
    var opts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    return m(iconBtn, _extends({}, opts, {
        parentClass: [CSS_CLASSES.block, opts.mini ? CSS_CLASSES.mini : null].join(' '),
        raised: true,
        ripple: {
            center: true,
            opacityDecayVelocity: 0.24
        },
        shadow: {
            increase: 5
        },
        ink: true,
        wash: true,
        animateOnTap: true
    }));
};

var component = {
    view: function view(ctrl) {
        var opts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

        return createView(ctrl, opts);
    }
};

export default component;