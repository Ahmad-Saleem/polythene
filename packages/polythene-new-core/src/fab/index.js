import { createProps, createContent, theme, element } from "./fab";
import classes from "./classes";
import vars from "./theme/vars";

export const fab = {
  createProps, createContent, theme, element,
  classes,
  vars
};
