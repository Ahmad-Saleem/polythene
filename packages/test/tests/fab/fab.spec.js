/* global describe, it, expect */
import m from "mithril";
import { tidy, defaultHtmlTidyOptions } from "mithril-jest";
import { tests } from "./tests.js";
import { runSnapshots } from "../../scripts/snapshots";
import fab from "polythene-fab";

runSnapshots(tests);

describe("FAB component", () => {
  it("no options", () => {
    const cmp = m(fab);
    const html = tidy(cmp, {
      ...defaultHtmlTidyOptions,
      wrap: false,
      indent: false
    });
    expect(html).toContain("<a tabindex=\"0\" class=\"pe-fab ");
    expect(html).toContain("<div class=\"pe-fab__content\"></div>");
  });
  it("option id", () => {
    const cmp = m(fab, {id: "id-x"});
    const html = tidy(cmp);
    expect(html).toContain("id=\"id-x\"");
  });
  it("option class", () => {
    const cmp = m(fab, {class: "class-x"});
    const html = tidy(cmp);
    expect(html).toContain("class-x\">");
  });
  it("option element", () => {
    const cmp = m(fab, {element: "span"});
    const html = tidy(cmp);
    expect(html).toContain("<span ");
    expect(html).toContain("</span>");
  });
  it("option tabindex", () => {
    const cmp = m(fab, {tabindex: 3});
    const html = tidy(cmp);
    expect(html).toContain("tabindex=\"3\"");
  });
  it("option before", () => {
    const cmp = m(fab, {before: m("span", "Before")});
    const html = tidy(cmp);
    expect(html).toContain("<span>Before</span>");
  });
  it("option after", () => {
    const cmp = m(fab, {after: m("span", "After")});
    const html = tidy(cmp);
    expect(html).toContain("<span>After</span>");
  });
});
