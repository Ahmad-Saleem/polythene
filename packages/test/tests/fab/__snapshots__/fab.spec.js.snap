exports[`test Child node 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Colored icon 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div style=\"color: rgb(51, 51, 51);\" class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Dark tone class + light theme class 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px;\" class=
\"pe-light-tone\">
  <a tabindex=\"0\" class=\"pe-fab\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-fab__content\">
      <div class=\"pe-icon pe-icon--regular\">
        <div class=\"pe-svg\">
          <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\"
          height=\"24\" viewbox=\"0 0 24 24\">
            <path d=
            \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div></a><a tabindex=\"0\" class=\"pe-fab tests-fab-themed-fab\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-fab__content\">
      <div class=\"pe-icon pe-icon--regular\">
        <div class=\"pe-svg\">
          <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\"
          height=\"24\" viewbox=\"0 0 24 24\">
            <path d=
            \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div></a>
</div>
"
`;

exports[`test Dark tone class + light tone 1`] = `
"<div style=\"background: rgb(255, 255, 255); padding: 10px;\">
  <a tabindex=\"0\" class=\"pe-fab\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-fab__content\">
      <div class=\"pe-icon pe-icon--regular\">
        <div class=\"pe-svg\">
          <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\"
          height=\"24\" viewbox=\"0 0 24 24\">
            <path d=
            \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div></a><a tabindex=\"0\" class=\"pe-fab tests-fab-themed-fab\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-fab__content\">
      <div class=\"pe-icon pe-icon--regular\">
        <div class=\"pe-svg\">
          <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\"
          height=\"24\" viewbox=\"0 0 24 24\">
            <path d=
            \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div></a>
</div>
"
`;

exports[`test Option: content 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-icon pe-icon--regular\">
    <div class=\"pe-svg\">
      <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
      \"24\" viewbox=\"0 0 24 24\">
        <path d=
        \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
        </path>
      </svg>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: icon -- dark theme class 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: icon 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: mini 1`] = `
"<a tabindex=\"0\" class=\"pe-fab pe-fab--mini\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: style (colors) 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div style=
\"color: rgb(255, 204, 188); background-color: rgb(78, 52, 46);\"
class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: url 1`] = `
"<a tabindex=\"0\" href=\"#!/shadow\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: z (0) 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-0\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-0\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Option: z (5) 1`] = `
"<a tabindex=\"0\" class=\"pe-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-5\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-5\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Themed FAB (colors) 1`] = `
"<a tabindex=\"0\" class=\"pe-fab tests-fab-themed-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Themed FAB -- dark theme class 1`] = `
"<a tabindex=\"0\" class=\"pe-fab tests-fab-themed-fab\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=
        \"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;
